src/tool_application_support.cpp

```diff
@@ -24,23 +25,62 @@
 
 namespace clangmetatool {
 
+namespace {
+
+#ifndef CLANG_INSTALL_LOCATION
+#define CLANG_INSTALL_LOCATION ""
+#endif
+
+std::string getExecutablePath(const std::string &argv0) {
+  // Use the address of 'main' to locate the executable name, it is possible
+  // that this may return an empty address
+  std::string exePath =
+      llvm::sys::fs::getMainExecutable(argv0.c_str(), nullptr);
+  if (!exePath.empty()) {
+    clang::DiagnosticsEngine diagnostics(new clang::DiagnosticIDs,
+                                 new clang::DiagnosticOptions,
+                                 new clang::IgnoringDiagConsumer);
+    clang::driver::Driver driver(
+        exePath, llvm::sys::getDefaultTargetTriple(),
+        diagnostics);
+    if (llvm::sys::fs::exists(driver.ResourceDir)) {
+      return exePath;
+    }
+  }
+
+  // Fall back to locating it on $PATH
+  const std::string &exeFileName = llvm::sys::path::filename(argv0);
```

Comment:
I would use `argv0` as passed, not just the basename. The issue is that you might have multiple installations of a particular clang tool, and the wrong one might be in the PATH. I guess you can skip this if argv0 is absolute, eg: `llvm::sys::path::is_absolute(argv0)`